torch/testing/_internal/common_nn.py

```diff
@@ -5322,33 +5206,141 @@ def to_double(x):
                     assert_module_parameters_are(torch.cuda.HalfTensor, 0)  # type: ignore[attr-defined]
         torch.set_num_threads(num_threads)
 
-    def _get_target(self):
-        return self._get_arg('target', False)
+    def noncontiguize(self, obj):
```

Comment:
I thought your original idea was to couple ModuleInfo with SampleInputs that had a hierarchy like:

constructor args
  - sample input
  - sample input
  - sample input

If so, maybe it makes sense to define methods like "noncontiguous()" on the sample input class?